common/lib/xmodule/xmodule/modulestore/split_mongo/mongo_connection.py

```diff
@@ -222,33 +222,37 @@ def __init__(self):
         except InvalidCacheBackendError:
             self.no_cache_found = True
 
-    def get(self, key):
+    def get(self, key, course_context=None):
         """Pull the compressed, pickled struct data from cache and deserialize."""
         if self.no_cache_found:
             return None
 
-        compressed_pickled_data = self.cache.get(key)
-        if compressed_pickled_data is None:
-            return None
-        return pickle.loads(zlib.decompress(compressed_pickled_data))
+        with TIMER.timer("CourseStructureCache.get", course_context) as tagger:
+            compressed_pickled_data = self.cache.get(key)
+            tagger.tag(from_cache=str(compressed_pickled_data is not None).lower())
+
+            if compressed_pickled_data is None:
+                # Always log cache misses, because they are unexpected
+                tagger.sample_rate = 1
+                return None
+            tagger.measure('compressed_size', len(compressed_pickled_data))
+
+            return pickle.loads(zlib.decompress(compressed_pickled_data))
 
-    def set(self, key, structure):
+    def set(self, key, structure, course_context=None):
         """Given a structure, will pickle, compress, and write to cache."""
         if self.no_cache_found:
             return None
 
-        pickled_data = pickle.dumps(structure, pickle.HIGHEST_PROTOCOL)
-        # 1 = Fastest (slightly larger results)
-        compressed_pickled_data = zlib.compress(pickled_data, 1)
+        with TIMER.timer("CourseStructureCache.set", course_context) as tagger:
+            pickled_data = pickle.dumps(structure, pickle.HIGHEST_PROTOCOL)
+            # 1 = Fastest (slightly larger results)
+            compressed_pickled_data = zlib.compress(pickled_data, 1)
 
-        # record compressed course structure sizes
-        dog_stats_api.histogram(
-            'compressed_course_structure.size',
-            len(compressed_pickled_data),
-            tags=[key]
-        )
-        # Stuctures are immutable, so we set a timeout of "never"
-        self.cache.set(key, compressed_pickled_data, None)
+            tagger.measure('compressed_size', len(compressed_pickled_data))
```

Comment:
How about capturing `uncompressed_size` for both get and set? If a large structure's compressed size ever gets too large for the cache, the first question we'll ask is "How big was the uncompressed structure?!?!?"